### PR TITLE
add UNRESTRICTED_CANNON_AIMING define

### PR DESCRIPTION
**Pull request by Reonu**
This define allows the player to aim in 360º with the cannon. Unsure on whether it should be on or not, I think so but feel free to discuss

### CODE REVIEW

File: src/game/mario_actions_automatic.c [hunk 1]
```diff
@@ -725,7 +725,11 @@ s32 act_in_cannon(struct MarioState *m) {
             marioObj->oMarioCannonInputYaw -= (s16)(m->controller->stickX * 10.0f);
 
             m->faceAngle[0] = CLAMP(m->faceAngle[0], 0, DEGREES(80));
+    #ifdef UNRESTRICTED_CANNON_AIMING
+            marioObj->oMarioCannonInputYaw = marioObj->oMarioCannonInputYaw;
+    #else
```
Comment (Arceveti) [inline src/game/mario_actions_automatic.c:730]:
```suggestion
    #ifndef UNRESTRICTED_CANNON_AIMING
```
Why not just this?

**Review by Arceveti — CHANGES_REQUESTED** [review]

File: src/game/mario_actions_automatic.c (hunk 1 above)
Comment (Reonu) [reply src/game/mario_actions_automatic.c:730]: either this makes no sense or I'm not understanding

you need the code in the ifdef for unrestricted cannon aiming and the code in the else for restricted cannon aiming, this wouldn't work

Comment (Arceveti) [reply src/game/mario_actions_automatic.c:730]: It's just setting `marioObj->oMarioCannonInputYaw` to itself.

Comment (axollyon) [reply src/game/mario_actions_automatic.c:730]: 500 iq code

(might also be the case that that's a typo and it's actually supposed to set something else)

Comment (Reonu) [reply src/game/mario_actions_automatic.c:730]: OH MY GOD

File: include/config/config_objects.h [hunk 2]
```diff
@@ -124,3 +124,12 @@
  * The number of segments Wiggler has, not including the head. Vanilla is 4.
  */
 #define WIGGLER_NUM_SEGMENTS     4
+
+/*****************
+ * -- CANNONS --
+ *****************/
+
+/**
+ * If you enable this define, you will be able to aim freely in 360 degrees with cannons
```
Comment (Arceveti) [inline include/config/config_objects.h:133]:
```suggestion
 * Removes the restriction on horizontal turning when aiming a cannon, allowing full 360 degree rotation.
```
Changed wording to be consistent with the other comments.

File: src/game/mario_actions_automatic.c [hunk 3]
```diff
@@ -725,7 +725,9 @@ s32 act_in_cannon(struct MarioState *m) {
             marioObj->oMarioCannonInputYaw -= (s16)(m->controller->stickX * 10.0f);
 
             m->faceAngle[0] = CLAMP(m->faceAngle[0], 0, DEGREES(80));
+    #ifndef UNRESTRICTED_CANNON_AIMING
             marioObj->oMarioCannonInputYaw = CLAMP(marioObj->oMarioCannonInputYaw, -0x4000, 0x4000);
```
Comment (Arceveti) [inline src/game/mario_actions_automatic.c:729]:
```suggestion
            marioObj->oMarioCannonInputYaw = CLAMP(marioObj->oMarioCannonInputYaw, -DEGREES(90), DEGREES(90);
```
Might as well do this too. Non-blocking.

**Review by Arceveti — APPROVED** [review]
formatting changes :)

**Comment by gheskett** [comment]
I think this might be better as (either replaced by or in addition to) an additional cannon bparam for rotational radius. A value of 0 would either exhibit vanilla or global defined behavior for compatibility reasons. A value of 1 should represent no rotation at all (who needs exactly 1° of adjustment anyway). And any value 180 or above would represent full rotational capacity. Clamping alone might not work for this idea since I assume cannons can rotate more than 1° per frame.

File: src/game/mario_actions_automatic.c [hunk 4]
```diff
@@ -725,7 +725,9 @@ s32 act_in_cannon(struct MarioState *m) {
             marioObj->oMarioCannonInputYaw -= (s16)(m->controller->stickX * 10.0f);
 
             m->faceAngle[0] = CLAMP(m->faceAngle[0], 0, DEGREES(80));
-            marioObj->oMarioCannonInputYaw = CLAMP(marioObj->oMarioCannonInputYaw, -0x4000, 0x4000);
```
Comment (gheskett) [inline src/game/mario_actions_automatic.c:728]: Actually wait, I think this variable is an s32, so it actually needs to be clamped no matter what. This is actually good news for the bparam idea though.

**Review by gheskett — CHANGES_REQUESTED** [review]

**Pull request closed without merging**